airflow/dag_processing/manager.py

```diff
@@ -576,8 +585,8 @@ def _run_parsing_loop(self):
         while True:
             loop_start_time = time.monotonic()
             ready = multiprocessing.connection.wait(self.waitables.keys(), timeout=poll_time)
-            if self._job:
-                self._job.heartbeat()
+            if hasattr(self._job_runner, "job") and self._job_runner.job is not None:
```

Comment:
```suggestion
            if getattr(self._job_runner, "job", None) is not None:
```

What’s the difference between `self._job_runner` not having the `job` attribute, and the attribute being None though? It seems to me the attribute can always be there and default to None.

Reply:
Ah that may make `self.job` awkward in runner classes. I think that can be improved with Generic but that could be an overkill. A comment here to briefly describe what the lack of the attribute indicates should be good enough.

Reply:
Yep. Having the job defined in generic runner does not fly as it would have to be added (really) in an __init__ but on the other hand we want to assing the runer to a job. This might be better addressed later, after the #30376 where job does not have job_runner reference any more. I will take a close look.

Reply:
I also added comment about it.